controller/controller_server/sync_lock.py

```diff
@@ -0,0 +1,50 @@
+import threading
+from collections import defaultdict
+
+from controller.common.csi_logger import get_stdout_logger
+from controller.controller_server.errors import ObjectAlreadyProcessingError
+
+logger = get_stdout_logger()
+
+ids_in_use = defaultdict(set)
+
+
+def add_to_ids_in_use(lock_key, object_id):
+    ids_in_use[lock_key].add(object_id)
+
+
+class SyncLock:
+    def __init__(self, lock_key, object_id, action_name):
+        self._lock = threading.Lock()

```

Comment:
I think the lock should also be "static", so the same lock would be shared between all threads, same as the dict it protects

Reply:
done